### DISCUSSION TITLE
"This model's maximum context length is 8192 tokens"

### DISCUSSION THREAD

**Discussion by DamianMehers** [community/community#56619] [Copilot Conversations]
### Select Topic Area

Bug

### Body

Hi,

I asked copilot chat it explain why I hit a breakpoint and the GitHub Copilot Chat Output Window gave this message:
_[Conversations Verbose] [KernelLogger] Raw response from Copilot: CopilotClientException:invalid_request_error:context_length_exceeded:This model's maximum context length is 8192 tokens. However, you requested 8221 tokens (4221 in the messages, 4000 in the completion). Please reduce the length of the messages or completion.
[Conversations Information] [KernelLogger] Token stream histogram_

I'm using VS Preview on Mac, latest updates installed.  Here is the complete info:
_VS info:
Microsoft Visual Studio Community 2022
Version 17.7.0 Preview 1.0
VisualStudio.17.Preview/17.7.0-pre.1.0+33711.374
Microsoft .NET Framework
Version 4.8.09037

Installed Version: Community

Visual C++ 2022   00482-90000-00000-AA072
Microsoft Visual C++ 2022

[http://ASP.NET](https://t.co/eAAVUvaIgO) and Web Tools   17.7.82.47043
[http://ASP.NET](https://t.co/eAAVUvaIgO) and Web Tools

Azure App Service Tools v3.0.0   17.7.82.47043
Azure App Service Tools v3.0.0

C# Tools   4.7.0-1.23226.11+6544980c8335ce25d8959afd4b6ee308686fed53
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.4.22 (main@fc8e91f)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

NuGet Package Manager   6.7.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit [https://docs.nuget.org](https://t.co/G3x133caEl)

Razor ([http://ASP.NET](https://t.co/eAAVUvaIgO) Core)   17.6.0.2326103+839fdd5e019d9e52a1f1985b54937a084f086b8b
Provides languages services for [http://ASP.NET](https://t.co/eAAVUvaIgO) Core Razor.

SQL Server Data Tools   17.6.13.0
Microsoft SQL Server Data Tools

TypeScript Tools   17.0.20426.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.7.0-1.23226.11+6544980c8335ce25d8959afd4b6ee308686fed53
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.7.0-beta.23221.1+858e2df9631a385cbddef62eeb424f9b3071b031
Microsoft Visual F# Tools

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

Visual Studio Tools for Unity   17.6.1.0
Visual Studio Tools for Unity

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

VSPackage Extension   1.0
VSPackage Visual Studio Extension Detailed Info

Xamarin   17.7.0.37 (main@5d96488)
Visual Studio extension to enable development for Xamarin.iOS and [http://Xamarin.Android](https://t.co/sxRNthleIG).

Xamarin Designer   17.7.1.10 (remotes/origin/main@4bc119239d)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

[http://Xamarin.Android](https://t.co/sxRNthleIG) SDK   13.2.0.11 (d17-5/b372b80)
[http://Xamarin.Android](https://t.co/sxRNthleIG) Reference Assemblies and MSBuild support.
    Mono: 6dd9def
    Java.Interop: xamarin/java.interop/d17-5@149d70fe
    SQLite: xamarin/sqlite/3.40.1@68c69d8
    [http://Xamarin.Android](https://t.co/sxRNthleIG) Tools: xamarin/xamarin-android-tools/d17-5@ca1552d_

Thanks,
   Damian

**Comment by markw-t** [comment]
Hi @DamianMehers  - we would love to gather some more data about the case where you hit this.

We can do that most effectively via [Visual Studio's "report a problem"](https://learn.microsoft.com/en-us/visualstudio/ide/how-to-report-a-problem-with-visual-studio?view=vs-2022) which lets us gather more information efficiently and route for followup. 

Please can I request that you open a new report-a-problem report from inside the Visual Studio version where you saw this problem. Once you have created a report, it would be great if you can attach the logs from the session where you saw the issue when explaining a breakpoint to that report.

You can find the logs from all your sessions on your hard disk in the %TEMP%\VSConversationsDiagnosticLogs folder, sorting in reverse date order to find the latest log. Logs are written when the session is closed (when you close Visual Studio).

Thanks for your help!

Regards
Mark Wilson-Thomas [MSFT]
Program Manager, Visual Studio extension for GitHub Copilot

**Reply by DamianMehers** [reply]
Done https://developercommunity.visualstudio.com/t/Follow-on-from-GitHub-issue-56619-This/10385027

Let me know if you need anything else.

Regards,

Damian

**Comment by sbond777** [comment]
@markw-t is it still possible to receive verbose response with how many tokens over the limit the prompt was? For me it just responds `Oops, your response got filtered. Vote down if you think this shouldn't have happened.` even though my prompt is always WAY smaller than 8k tokens. In fact, it gets triggered for a ~1000 token prompt.
